### PR TITLE
Add locale ca_ES (Catalan)

### PR DESCRIPTION
**Pull request by aaronplanell**
Hello,

I added the local for ca_ES (Catalan). Therefore, this fork only have four new files:
```
components/calendar/locale/ca_ES.tsx
components/date-picker/locale/ca_ES.tsx
components/locale-provider/ca_ES.tsx
components/time-picker/locale/ca_ES.tsx
```

And a modified file:
```
components/locale-provider/__tests__/index.test.js
```

Previously to this fork, I updated to other projects:
- rc-calendar: https://github.com/react-component/calendar/pull/213
- rc-pagination: https://github.com/react-component/pagination/pull/59

Finally, I run:
```
antd-tools run ts-lint
antd-tools run compile
```

Everything it's OK. Waiting for your fork :smile:

### CODE REVIEW

File: components/locale-provider/ca_ES.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import moment from 'moment';
+moment.locale('ca');
+
+import Pagination from 'rc-pagination/lib/locale/ca_ES';
+import DatePicker from '../date-picker/locale/ca_ES';
+import TimePicker from '../time-picker/locale/ca_ES';
+import Calendar from '../calendar/locale/ca_ES';
+
+export default {
+  locale: 'en',
```
Comment (corneyl) [inline components/locale-provider/ca_ES.tsx]: Shouldn't this be `es`?

Comment (aaronplanell) [inline components/locale-provider/ca_ES.tsx]: sorry it's ca.

**Comment by aaronplanell** [comment]
For your information, previously, I had two discarded PR: [4927](https://github.com/ant-design/ant-design/pull/4927) and [4928](https://github.com/ant-design/ant-design/pull/4928).

File: components/locale-provider/ca_ES.tsx (hunk 1 above)
Comment (aaronplanell) [inline components/locale-provider/ca_ES.tsx]: I correct it now.

Comment (aaronplanell) [inline components/locale-provider/ca_ES.tsx]: Thanks a lot @corneyl ! I made all those changes three times, hehehe.

**Comment by aaronplanell** [comment]
Well, finally I think that all have worked :smile: . I suppose that this change will be available in the next release (2.8), is that correct? 

Thanks a lot for your patience, especially to @afc163 , @yesmeck, benjycui and @corneyl :+1:

**Review by corneyl — APPROVED** [review]

**Pull request merged**